src/index.ts

```diff
@@ -1,31 +1,75 @@
+import { parseOrbManifest } from "@circleci/circleci-config-parser";
 import { orb } from "@circleci/circleci-config-sdk";
 import { OrbImportManifest } from "@circleci/circleci-config-sdk/dist/src/lib/Orb/types/Orb.types";
 import { fetch } from "undici";
+import { parse } from "yaml";
 import OrbImport = orb.OrbImport;
 
+export interface OrbVersionRef {
+  namespace: string;
+  orb: string;
+  version: string;
+}
+
+export type OrbImportRef = { alias: string } & OrbVersionRef;
+
 /// Downloads the manifest for the given orb and generates the `OrbImport` for use in the SDK
 export async function importOrb({
   alias,
   namespace,
   orb,
   version,
-}: {
-  alias: string;
-  namespace: string;
-  orb: string;
-  version: string;
-}): Promise<OrbImport> {
-  const manifest = await fetchOrbManifest(`${namespace}/${orb}@${version}`);
-  return new OrbImport(alias, namespace, orb, version, "", manifest);
+}: OrbImportRef): Promise<OrbImport> {
+  const source = await fetchOrbSource({ namespace, orb, version });
+  const manifest = await generateOrbManifestFromSource(source);
+  return new OrbImport(alias, namespace, orb, version, undefined, manifest);
```

Comment:
I'm guessing you did `import OrbImport = orb.OrbImport` to avoid a name collision with the `orb` argument. 

You could change the `orb` property to `orbName` to avoid the `import OrbImport = orb.OrbImport`. 

```ts
export interface OrbVersionRef {
  namespace: string;
  orbName: string;
  version: string;
}
```
If it needs to be named `orb` for whatever reason, you could also try renaming the argument in `importOrb` from `orb` to `orbName`. This isn't that much better, but something to consider:
```ts
export async function importOrb({
  alias,
  namespace,
  orb: orbName,
  version,
}: OrbImportRef): Promise<OrbImport> {
  const source = await fetchOrbSource({ namespace, orb: orbName, version });
  const manifest = await generateOrbManifestFromSource(source);
  return new orb.OrbImport(alias, namespace, orbName, version, undefined, manifest);
}
```

Reply:
Good to know that I can do that! Thanks!